src/Microsoft.TestPlatform.CommunicationUtilities/LengthPrefixCommunicationChannel.cs

```diff
@@ -63,6 +63,13 @@ public Task Send(string data)
         /// <inheritdoc />
         public Task NotifyDataAvailable()
         {
+            // TODO: Review the comment below, because it says something different than what is
+            // actually happening, and doing what it suggests would potentially lose messages.
+            // For example in the case where we start testhost process, send it version, and
+            // it responds, we then replace the handler with a new one, and there is quite a long time
+            // (tens of milliseconds) when there is no handler present, which would pump the message
+            // and dump it.
```

Comment:
it will be consumed by poll after right?

Reply:
Not sure I get what you are asking. From how I understand how this works is that we poll, and invoke notification (NotifyDataAvailable) when there are data. 

Then in the notification handler (in the code here), we read the stream to get the message data, and send it to receivers by calling MessageReceived. 

If we poll, and read even where there are no subscribers to MessageReceived we just lose that message. 

Am I wrong?

Reply:
Nope it's true(like I understood at least), that "consumed by poll" was a bad way to say what you wrote.

Reply:
@Evangelink  for awareness on how this works.